src/Compute/Compute/Extension/AzureDiskEncryption/SetAzureDiskEncryptionExtension.cs

```diff
@@ -839,7 +839,10 @@ private Hashtable GetVmPublicSettingsForMigration(VirtualMachine virtualMachineR
         {
             Hashtable vmPublicSettings = JsonConvert.DeserializeObject<Hashtable>(adeExtensionInstanceView.PublicSettings);
             VerifyKeyVaultProperties(virtualMachineResponse, vmPublicSettings);
-            vmPublicSettings.Add(AzureDiskEncryptionExtensionConstants.migrateAdeOperationKey, AzureDiskEncryptionExtensionConstants.migrateAdeOperationValue);
+            if (!vmPublicSettings.Contains(AzureDiskEncryptionExtensionConstants.migrateAdeOperationKey))
+            {
+                vmPublicSettings.Add(AzureDiskEncryptionExtensionConstants.migrateAdeOperationKey, AzureDiskEncryptionExtensionConstants.migrateAdeOperationValue);
```

Comment:
Not sure I am following this. Some code comment might be helpful here. Does this result in a migrate even if not specified by the customer? If so then maybe also provide an informational message to the user at time of the change.

Reply:
`GetVmPublicSettingsForMigration()` is only called when Migrate or MigrateRecovery option is used. Hence I added an additional check here to avoid adding "MigrateFlag" to the publiSettings when the settings already has it i.e When the customer already tried migrate option before.

I can add a comment here as you suggested.